### PR TITLE
Add a bunch of expressions to the Helmholtz expression writer

### PR DESCRIPTION
**Pull request by eslickj**
This adds a bunch of expressions to the Helmholtz EoS expressions writer.  The point of the expression writer is to let you put things that look like thermo property function calls in your model without a state block.  The motivation for this addition is to make it easier to use the Helmholtz EoS in the general property framework to calculate pure component properties. 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/generic_models/properties/helmholtz/helmholtz.py [hunk 1]
```diff
@@ -342,6 +342,9 @@ class HelmholtzThermoExpressions(object):
     can be used to calculate any other property.  You can specify an integer
     value 1 or 0 for x with any state varaibles to get liquid or vapor
     properies.
+
+    The purpose of this class is to allow for straight forward calculation of
+    preoprties without a full state block for each set of conditions.
```
Comment (andrewlee94) [inline idaes/generic_models/properties/helmholtz/helmholtz.py:347]: Typo: properties

File: idaes/generic_models/properties/helmholtz/helmholtz.py [hunk 2]
```diff
@@ -363,7 +366,7 @@ def add_funcs(self, names=None):
 
     def basic_calculations(
             self, h=None, s=None, p=None, T=None, u=None, x=None):
-        """This function is called as this basis for most thermo expression
+        """This function is called as the basis for most thermo expression
         writer functions.  It takes the given state variables and returns
         expressions for liqid density, vapor density, vapor fraction and
```
Comment (andrewlee94) [inline idaes/generic_models/properties/helmholtz/helmholtz.py:371]: Typo: liquid

File: idaes/generic_models/properties/helmholtz/helmholtz.py [hunk 3]
```diff
@@ -436,36 +438,112 @@ def basic_calculations(
...
+        """Mixed phase entropy"""
         blk, delta_liq, delta_vap, tau, x = self.basic_calculations(**kwargs)
         self.add_funcs(names=["func_s"])
         s = blk.func_s(delta_liq, tau)*(1-x) + blk.func_s(delta_vap, tau)*x
         return s * self.param.uc_kJ_per_kgK_to_J_per_molK
 
+    def s_liq(self, **kwargs):
+        """Liquid phase entropy"""
+        blk, delta_liq, delta_vap, tau, x = self.basic_calculations(**kwargs)
+        self.add_funcs(names=["func_s"])
```
Comment (andrewlee94) [inline idaes/generic_models/properties/helmholtz/helmholtz.py:450]: Is there any issue with calling `add_funcs` multiple times with the same name? From what I can tell, `s`, `s_liq` and `s_Vap` would all call `self.add_funcs(names=["func_s"])`, so this code could be executed multiple times.

File: idaes/generic_models/properties/helmholtz/helmholtz.py [hunk 4]
```diff
@@ -981,8 +1151,7 @@ class HelmholtzStateBlockData(StateBlockData):
     """
 
     def initialize(self, *args, **kwargs):
-        # With this particualr property pacakage there is not need for
-        # initialization
+        # This particualr property pacakage has no need for initialization
```
Comment (andrewlee94) [inline idaes/generic_models/properties/helmholtz/helmholtz.py:1154]: Typos: particular, package

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
This looks OK as best I can tell, although there are a few areas where I didn't understand everything. A couple of typos and a question about potentially adding the same function multiple times.

File: idaes/generic_models/properties/helmholtz/helmholtz.py (hunk 3 above)
Comment (eslickj) [reply idaes/generic_models/properties/helmholtz/helmholtz.py:450]: It's ok.  That checks that the function doesn't exist before adding it.

**Review by andrewlee94 — APPROVED** [review]
Looks good.

File: idaes/generic_models/properties/tests/test_iapws95_basic.py [hunk 5]
```diff
@@ -211,92 +236,188 @@ def model_transport(self):
...
+                and value(abs(p - self.Pc) / self.Pc) < 0.05)
+                or value(T) < 281
+            ):
+                # some properties blow up or become very sensitive around the
+                # critical point, that will be broken out for special testing
+                # this test just tests that expressions work right, so we'll
+                # skip near-critical area.
                 continue
 
+            # The test tolerances aren't super thight here, some property
```
Comment (dallan-keylogic) [inline idaes/generic_models/properties/tests/test_iapws95_basic.py:282]: Typo

File: idaes/generic_models/properties/tests/test_iapws95_basic.py [hunk 6]
```diff
@@ -211,92 +236,188 @@ def model_transport(self):
...
+                assert value(te.s_vap(h=h, p=p)) == pytest.approx(value(s), rel=tol)
+
+            assert value(te.u(h=h, p=p)) == pytest.approx(value(u), rel=tol)
+            assert value(te.u(h=h, p=p, x=x)) == pytest.approx(value(u), rel=tol)
+            if x == 0:
+                assert value(te.u_liq(h=h, p=p)) == pytest.approx(value(u), rel=tol)
+            else:
+                assert value(te.u_vap(h=h, p=p)) == pytest.approx(value(u), rel=tol)
+
+            # To midigate the data sig figs calculate T and s from h and p for consistency
```
Comment (dallan-keylogic) [inline idaes/generic_models/properties/tests/test_iapws95_basic.py:312]: Mitigate

File: idaes/generic_models/properties/tests/test_iapws95_basic.py [hunk 7]
```diff
@@ -211,92 +236,188 @@ def model_transport(self):
...
+            if x == 0:
+                assert value(te.g_liq(h=h, p=p)) == pytest.approx(
+                    value(h - te.T(h=h, p=p) * te.s(h=h, p=p)), rel=tol
+                )
+            else:
+                assert value(te.g_vap(h=h, p=p)) == pytest.approx(
+                    value(h - te.T(h=h, p=p) * te.s(h=h, p=p)), rel=tol
+                )
+
+            # To midigate the data sig figs calculate u, T, and s from h and p for consistency
```
Comment (dallan-keylogic) [inline idaes/generic_models/properties/tests/test_iapws95_basic.py:328]: Mitigate

**Review by dallan-keylogic — COMMENTED** [review]
Few typos

**Review by dallan-keylogic — APPROVED** [review]

**Review by andrewlee94 — APPROVED** [review]
Looks good.

**Pull request merged**